### ISSUE TITLE
Latest clangd version is incorrect

### ISSUE THREAD

**Issue by comiluv** [williamboman/mason.nvim#1793]
### I've searched open issues for similar requests

- [X] Yes

### I've manually reviewed logs to find potential errors

- [X] Yes

### I've recently downloaded the latest plugin version of mason.nvim

- [X] Yes

### Problem description

Mason does not install latest version of clangd which is 19.1.0 and instead installs version 18.1.3
Also when forced to install version 19.1.0 using `:MasonInstall clangd@19.1.0`, Mason incorrectly reports updated version of clangd is available but is actually just 18.1.3

### Expected behavior

Mason correctly installs version 19.1.0 of clangd
Also Mason should correctly report that version 19.1.0 of clangd is the latest

### Steps to reproduce

1. `:MasonInstall clangd`
2. See that the version installed is 18.1.3

### Affected packages

clangd

### Neovim version (>= 0.7)

NVIM v0.10.1
Build type: Release
LuaJIT 2.1.1713484068

### Operating system/version

Windows 11 (Win32NT             10.0.22631.0 Microsoft Windows NT 10.0.22631.0)

### Healthcheck

```Text
==============================================================================
mason: require("mason.health").check()

mason.nvim ~
- OK mason.nvim version v1.10.0
- OK PATH: prepend
- OK Providers: 
    mason.providers.registry-api
    mason.providers.client
- OK neovim version >= 0.7.0

mason.nvim [Registries] ~
- OK Registry `github.com/mason-org/mason-registry version: 2024-09-20-friendly-knight` is installed.

mason.nvim [Core utils] ~
- WARNING unzip: not available
  - ADVICE:
    - spawn: unzip failed with exit code - and signal -. unzip is not executable
- WARNING wget: not available
  - ADVICE:
    - spawn: wget failed with exit code - and signal -. wget is not executable
- OK curl: `curl 8.8.0 (Windows) libcurl/8.8.0 Schannel zlib/1.3 WinIDN
`
- WARNING gzip: not available
  - ADVICE:
    - spawn: gzip failed with exit code - and signal -. gzip is not executable
- OK tar: `bsdtar 3.7.2 - libarchive 3.7.2 zlib/1.2.5.f-ipp liblzma/5.2.5 bz2lib/1.0.8 libzstd/1.5.4 
`
- OK pwsh: `7.4.5 Microsoft Windows 10.0.22631 Win32NT
`
- OK 7z: `7-Zip 24.08 (x64) : Copyright (c) 1999-2024 Igor Pavlov : 2024-08-11
`

mason.nvim [Languages] ~
- WARNING Go: not available
  - ADVICE:
    - spawn: go failed with exit code - and signal -. go is not executable
- WARNING Composer: not available
  - ADVICE:
    - spawn: composer.bat failed with exit code - and signal -. composer.bat is not executable
- WARNING PHP: not available
  - ADVICE:
    - spawn: php failed with exit code - and signal -. php is not executable
- WARNING luarocks: not available
  - ADVICE:
    - spawn: luarocks failed with exit code - and signal -. Could not find executable "luarocks" in path.
      
- WARNING Ruby: not available
  - ADVICE:
    - spawn: ruby failed with exit code - and signal -. ruby is not executable
- WARNING RubyGem: not available
  - ADVICE:
    - spawn: gem.cmd failed with exit code - and signal -. gem.cmd is not executable
- OK node: `v20.16.0
`
- WARNING julia: not available
  - ADVICE:
    - spawn: julia failed with exit code - and signal -. julia is not executable
- OK python: `Python 3.12.6
`
- OK cargo: `cargo 1.81.0 (2dbb1af80 2024-08-20)`
- OK JAVA_HOME: `java version "1.8.0_421"
`
- OK java: `java version "1.8.0_421"
`
- OK npm: `10.8.1`
- OK javac: `javac 21.0.2
`
- OK pip: `pip 24.2 from C:\Users\windowsUser\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip (python 3.12)

`
- OK python venv: `Ok`

mason.nvim [GitHub] ~
- OK GitHub API rate limit. Used: 0. Remaining: 5000. Limit: 5000. Reset: 2024-09-20 오후 1:51:46.
```

### Screenshots or recordings

<img width="588" alt="Neovide 2024-09-20 오후 12_45_40" src="https://github.com/user-attachments/assets/643e13d0-8c63-4365-926f-ac972f38f976">

**Comment by Fabian-programmer** [comment]
`:MasonInstall clangd@19.1.0` worked for me.

 19.1.0 was a pre-relese until, like yesterday.

**Comment by comiluv** [comment]
> `:MasonInstall clangd@19.1.0` worked for me.
> 
> 19.1.0 was a pre-relese until, like yesterday.

[stable 19.1.0 was released two weeks ago](https://github.com/clangd/clangd/releases/tag/19.1.0) (Sep 19, 2024 4:16PM GMT+9)
AFAIK Mason didn't recognize 19.1.0 until today but it's fixed now

**Comment by Fabian-programmer** [comment]
https://github.com/clangd/clangd/issues/2127#issuecomment-2384757214

**Comment by comiluv** [comment]
> [clangd/clangd#2127 (comment)](https://github.com/clangd/clangd/issues/2127#issuecomment-2384757214)

What is your point? 19.1.0 was released 'after' 18.1.3. Being a pre-release doesn't mean the release is `older`
 Mason reporting "new version available 19.1.0 -> 18.1.3" is clearly an error, because it's not "new version"(sic)

**Comment by Fabian-programmer** [comment]
My point is that you should start reading the docs. This was never an issue.
19.1.0 was a pre-release until today, see clangd comment I pasted. After that the registry got updated.

https://github.com/mason-org/mason-registry/commit/9c5e7f360d982eb11c0aeafeeb66ae7dc1486563

**Comment by comiluv** [comment]
I wrote in the original post and also clarified in the comment above that `Mason reporting "new version available 19.1.0 -> 18.1.3" is clearly an error, because it's not "new version"(sic)`

> My point is that you should start reading the docs. 
> 19.1.0 was a pre-release until today,

Your comment is not only unrelated to the original post but wrong. Where in the mason.nvim's docs say "Older releases not marked as prerelease are considered newer versions than releases marked as prelease?" It's clear that *you* are the one who should start reading what people actually post.

Also you're showing lots of weird hostility in comments for a simple versioning issue report. Whatever your problem is, keep it to your self.

**Comment by comiluv** [comment]
Here's how versioning scheme works for reference:
https://semver.org/#spec-item-11
